### ISSUE TITLE
Remove doc/words/examples/aimbot.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4007]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ivank reported|
|-|-|
|Trac ID|trac#4007|
|Type|defect|
|Created|2009-09-06 19:26:14Z|
|Branch|https://github.com/twisted/twisted/tree/remove-aimbot-4007|

aimbot.py doesn't work. If you run it with logging, you'll see something like this:

```
2009-09-06 12:23:56-0700 [-] Log opened.
2009-09-06 12:23:56-0700 [-] aimbot.py:14: exceptions.DeprecationWarning: twisted.words.protocols.toc is deprecated since Twisted 9.0.  Use twisted.words.protocols.oscar instead.
2009-09-06 12:23:57-0700 [-] /opt/Python-latest/lib/python2.7/site-packages/twisted/words/im/__init__.py:8: exceptions.UserWarning: twisted.im will be undergoing a rewrite at some point in the future.
2009-09-06 12:23:57-0700 [Uninitialized] connection made! <<class 'twisted.internet.tcp.Client'> to ('toc.oscar.aol.com', 9898) at 16717d0>
2009-09-06 12:23:57-0700 [aimBot,client] 
                                           MYUSERNAME
2009-09-06 12:23:57-0700 [aimBot,client] toc_signon login.oscar.aol.com 5190  MYUSERNAME 0x161e0b4d20169760 english "penguin"
```

The account never appears online.

twisted.words.protocols.toc was deprecated in [#3580](https://github.com/twisted/twisted/issues/3580).

Attachments:

* [mypatch.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/55b/55ba0aaca3bc93c0ad17e87b33c3446de2e01047/9b1748c285b9beb56f91af478872472c3811f615.patch) (1940 bytes) - added by Poincare on 2009-11-21 02:35:22Z - review
* [removeaimbot-4007.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/55b/55ba0aaca3bc93c0ad17e87b33c3446de2e01047/fd52bd12b95259931db60b6d5beebeff80e17e59.diff) (2617 bytes) - added by djfroofy on 2010-02-22 23:09:41Z - Remove aimbot.py example and ref in docs

<details><summary>Searchable metadata</summary>

```
trac-id__4007 4007
type__defect defect
reporter__ivank ivank
priority__low low
milestone__ 
branch__branches_remove_aimbot_4007 branches/remove-aimbot-4007
branch_author__djroofy__thijs djroofy, thijs
status__closed closed
resolution__fixed fixed
component__words words
keywords__documentation documentation
time__1252265174000000 1252265174000000
changetime__1295606991000000 1295606991000000
version__None None
owner__ 
cc__thijs cc__exarkun cc__njoyce cc__djfroofy
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

What about rewriting it with twisted.words.protocols.oscar instead?

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|ivank commented|
|-|-|

oscardemo.py already does that, I think.

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ivank removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to ivank|
|-|-|

I think this is premature.  `twisted.words.protocols.toc` hasn't been deprecated in any release yet.  It's probably useless, but the example should stay until the code itself is removed.  That would be in 9.0+2.

**Comment by twisted-trac** [comment]
|<img alt="njoyce's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|njoyce commented|
|-|-|

Now that [#3580](https://github.com/twisted/twisted/issues/3580) is fixed and `twisted.words.protocols.toc` is deprecated, the attached patch appears to do the job.

**Comment by twisted-trac** [comment]
|<img alt="njoyce's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|njoyce removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)|@djfroofy set owner to @djfroofy<br>@djfroofy set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)<a name="note_8"></a>|@djfroofy set owner to @exarkun<br>@djfroofy set status to `new`|
|-|-|

Updated a patch with reference removed in documentation.  If `twisted.words.protocols.toc` was supposed to removed in 9.0+2 maybe it's due time to really remove it considering current 10.0 release?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

We need to wait a bit longer on this.  9.0 was released in December, three months ago.  We can think about this for the release around December 2010.

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_11"></a>|amacleod commented|
|-|-|

This ticket depends on [#4363](https://github.com/twisted/twisted/issues/4363) (remove TOC)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra commented|
|-|-|

(In [30527]) Branching to 'remove-aimbot-4007'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra commented|
|-|-|

(In [30528]) Apply removeaimbot-4007.diff. refs [#4007](https://github.com/twisted/twisted/issues/4007)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra commented|
|-|-|

(In [30529]) add news file. refs [#4007](https://github.com/twisted/twisted/issues/4007)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_16"></a>|@cyli set owner to @thijstriemstra|
|-|-|

This only seems to remove the content of the aimbot.py file.  Could you remove the file itself?

Otherwise looks good to merge, as long as the documentation builder still works once the file is deleted.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_17"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Thanks for the review.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_18"></a>|@thijstriemstra commented|
|-|-|

(In [30538]) Actually remove file, refs [#4007](https://github.com/twisted/twisted/issues/4007)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_19"></a>|@thijstriemstra commented|
|-|-|

Replying to [cyli](#note_16):
> This only seems to remove the content of the aimbot.py file.  Could you remove the file itself?
> 
> Otherwise looks good to merge, as long as the documentation builder still works once the file is deleted.

Results for the [documentation builder](http://buildbot.twistedmatrix.com/builders/documentation/builds/883). I'll merge it after [#4363](https://github.com/twisted/twisted/issues/4363) landed.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_20"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [30544]) Merge remove-aimbot-4007: Removed the non-working AIM bot example, depending on the obsolete
twisted.words.protocols.toc functionality.

Author: thijs, djroofy
Reviewer: cyli
Fixes: [#4007](https://github.com/twisted/twisted/issues/4007)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
